snapcraft/commands/lint.py

```diff
@@ -107,3 +120,73 @@ def _get_assert_file(self, snap_file: Path) -> Optional[Path]:
 
         emit.debug(f"Found assertion file {str(assert_file)!r}.")
         return assert_file
+
+    def _prepare_instance(
+        self,
+        snap_file: Path,
+        assert_file: Optional[Path],
+        http_proxy: Optional[str],
+        https_proxy: Optional[str],
+    ) -> None:
+        """Prepare an instance to lint a snap file.
+
+        Launches an instance, pushes the snap file and optional assert file, then
+        re-executes `snapcraft lint` inside the instance.
+
+        :param snap_file: Path to snap file to push into the instance.
+        :param assert_file: Optional path to assertion file to push into the instance.
+        :param http_proxy: http proxy to add to environment
+        :param https_proxy: https proxy to add to environment
+        """
+        emit.progress("Checking build provider availability.")
+
+        provider = providers.get_provider()
+        providers.ensure_provider_is_available(provider)
+
+        # create base configuration
+        instance_name = "snapcraft-linter"
+        build_base = providers.SNAPCRAFT_BASE_TO_PROVIDER_BASE["core22"]
+        base_configuration = providers.get_base_configuration(
+            alias=build_base,
+            instance_name=instance_name,
+            http_proxy=http_proxy,
+            https_proxy=https_proxy,
+        )
+
+        emit.progress("Launching instance.")
+
+        with provider.launched_environment(
+            project_name="snapcraft-linter",
+            project_path=Path().absolute(),
+            base_configuration=base_configuration,
+            build_base=build_base.value,
+            instance_name=instance_name,
+            allow_unstable=False,
+        ) as instance:
+            # push snap file
+            snap_file_instance = Path("/root") / snap_file.name
```

Comment:
There's a `get_managed_environment_home_path()` in os_utils that's used during regular packing operations. We could use it here too.

Reply:
Thanks, updated!